packages/xod-project/src/Traversing.re

```diff
@@ -0,0 +1,145 @@
+open Belt;
+
+type pinWithOwnerNode = (Pin.t, Node.t);
+
+/*
+==============================================================================================================
+  Utility functions
+==============================================================================================================
+*/
+
+let isCompositionPatch = patch => (
+  !Patch.isNotImplementedInXod(patch) &&
+  !Patch.isTerminal(patch)
+);
+
+let getUpstreamPinPairFromInputPin: (Pin.t, Node.t, Patch.t, Project.t) => option(pinWithOwnerNode) =
```

Comment:
I suggest to get rid of "PinPair" term, since it's called "PinWithOwnerNode" now